### ISSUE TITLE
fix (rather hacky) for MIN/MAX collision

### ISSUE THREAD

**Issue by edscott** [OPM/opm-common#2027]
patch for collision with MIN/MAX macros:
[UDQFunction.hpp.patch.txt](https://github.com/OPM/opm-common/files/5395998/UDQFunction.hpp.patch.txt)

Showstopper:

In file included from /usr/home/edscott/src/OPM/opm-common/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQFunctionTable.hpp:26:
/usr/home/edscott/src/OPM/opm-common/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQFunction.hpp:58:40: error: too few arguments
provided to function-like macro invocation
static UDQSet MIN(const UDQSet& arg);
^
/usr/include/sys/param.h:305:9: note: macro 'MIN' defined here
#define MIN(a,b) (((a)<(b))?(a):(b))
^
In file included from /usr/home/edscott/src/OPM/opm-simulators/flow/flow_ebos_gasoil.cpp:25:
In file included from /usr/home/edscott/src/OPM/opm-simulators/opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp:26:
In file included from /usr/home/edscott/src/OPM/opm-simulators/opm/simulators/flow/BlackoilModelEbos.hpp:27:
In file included from /usr/home/edscott/src/OPM/opm-simulators/ebos/eclproblem.hh:52:
In file included from /usr/home/edscott/src/OPM/opm-simulators/ebos/eclcpgridvanguard.hh:30:
In file included from /usr/home/edscott/src/OPM/opm-simulators/ebos/eclbasevanguard.hh:52:
In file included from /usr/home/edscott/src/OPM/opm-common/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQConfig.hpp:29:
In file included from /usr/home/edscott/src/OPM/opm-common/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQDefine.hpp:31:
In file included from /usr/home/edscott/src/OPM/opm-common/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQFunctionTable.hpp:26:
/usr/home/edscott/src/OPM/opm-common/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQFunction.hpp:59:40: error: too few arguments
provided to function-like macro invocation
static UDQSet MAX(const UDQSet& arg);
^
/usr/include/sys/param.h:306:9: note: macro 'MAX' defined here
#define MAX(a,b) (((a)>(b))?(a):(b))
^
1 warning and 2 errors generated.

**Comment by joakim-hove** [comment]
Thank you for the patch - problem understood, might push a different fix.

**Comment by edscott** [comment]
Yeah, I would  avoid using all caps for functions in the class, since usually all caps are used for macros.
